app/src/androidTest/java/org/kiwix/kiwixmobile/settings/KiwixSettingsActivityTest.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.settings
+
+import org.junit.Test
+import org.kiwix.kiwixmobile.BaseActivityTest
+
+class KiwixSettingsActivityTest : BaseActivityTest<KiwixSettingsActivity>() {
+
+  override var activityRule = activityTestRule<KiwixSettingsActivity>()
+
+  @Test
+  fun testSettingsActivity() {
+    settingsRobo {
+      checkRemainingTextViews()
```

Comment:
so this api needs a bit of work to align with idea of a robot. It is supposed to be 1 user interaction to 1 method and also optional assert methods eg instead of `invokeStorageDialog()`

```
clickOnStoragePreference()
assertStorageDialogDisplayed()
dismissDialog()
```

If you want more detail those links I posted should be fairly in depth

Reply:
Gotcha!

Reply:
But doing so will lead to creation of many smaller either one or two liner functions in robot. Would that be okay, due to this reason I didn't make 'em all

Reply:
`checkRemainingTextViews()` && `toggleButtons()` are still too broad

Reply:
Segregating the functions for testing each single togglable button and textView on the screen